### PR TITLE
refactor: Clean up, match presto, and optimize ipprefix string -> ipprefix

### PR DESCRIPTION
**Pull request by mohsaka**
Clean up the code for converting from string->ipprefix.
Also removed all unnecessary error message processing to match java presto's output.
Removed unnecessary variables
Enable IPPREFIX cast test

### CODE REVIEW

**Comment by Yuhta** [comment]
@yuandagits

File: velox/functions/prestosql/types/IPPrefixType.h [hunk 1]
```diff
@@ -97,55 +48,41 @@ tryParseIpPrefixString(folly::StringPiece ipprefixString) {
...
+        threadSkipErrorDetails()
+            ? Status::UserError()
+            : Status::UserError(
+                  "Cannot cast value to IPPREFIX: {}", ipprefixString));
   }
 
+  folly::ByteArray16 addrBytes;
   int128_t intAddr;
-  // Similar to IPAdressType, assume Velox is always on little endian systems
+  addrBytes = (ip.isIPv4Mapped() || ip.isV4())
```
Comment (czentgr) [inline velox/functions/prestosql/types/IPPrefixType.h:77]: You can combine line 75 and this one together and declare and initialize addrBytes.
And you can use `auto` for the type too.

Also you'd ant to move the declaration of `intAddr` before line 82. Best to initialize it to 0 too.

**Review by yuandagits — APPROVED** [review]
look good, thanks!

@czentgr has good suggestion re. the simplification

File: velox/functions/prestosql/types/IPPrefixType.h (hunk 1 above)
Comment (mohsaka) [reply velox/functions/prestosql/types/IPPrefixType.h]: Done!

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**